core/embed/trezorhal/optiga/optiga.c

```diff
@@ -363,8 +371,64 @@ static int optiga_pin_init_stretch(void) {
   return OPTIGA_SUCCESS;
 }
 
-static int optiga_pin_stretch_secret(OPTIGA_UI_PROGRESS ui_progress,
-                                     uint8_t secret[OPTIGA_PIN_SECRET_SIZE]) {
+static int optiga_pin_stretch_common(
+    OPTIGA_UI_PROGRESS ui_progress, HMAC_SHA256_CTX *ctx,
+    const uint8_t input[OPTIGA_PIN_SECRET_SIZE], bool version4) {
+  // Implements the functionality that is common to optiga_pin_stretch_secret()
+  // and the legacy function optiga_pin_stretch_secret_v4().
+
+  uint8_t buffer[ENCRYPT_SYM_PREFIX_SIZE + OPTIGA_PIN_SECRET_SIZE] = {0};
+  size_t size = 0;
+
+  // Combine intermediate result with OID_PIN_CMAC.
+  optiga_result res =
+      optiga_encrypt_sym(OPTIGA_SYM_MODE_CMAC, OID_PIN_CMAC, input,
+                         OPTIGA_PIN_SECRET_SIZE, buffer, sizeof(buffer), &size);
```

Comment:
I would either
  * check `size` here and in many other places or,
  * rewrite the `optiga_*` functions so they can accept `NULL`.

Reply:
I'm not sure what you mean by "`&size` can't be `NULL`" or why it is relevant. What is relevant is that `size_t *output_size` can be `NULL`.

Reply:
"`&size` can't be `NULL`", that's a fact. Thus checking that it is not `NULL` makes no sense. Furthermore, since "`&size` can't be `NULL`" and `&size` is passed here as the `output_size` argument to `optiga_encrypt_sym()`, it follows that `output_size` can't be `NULL` in this particular call.

Reply:
We discussed it in person, but we did not reach a consensus.